### PR TITLE
dev-util/lcov: Add gcc-9 support in 1.14-r1

### PR DESCRIPTION
**Pull request by gekola**
This adds upstream patches needed for gcc-9 support to 1.14 since upstream [does not][1] plan on making new releases soon and other distros seem to have included relevant changes in their 1.14 packages ([Debian][2], [Fedora][3]).

[1]: https://github.com/linux-test-project/lcov/issues/58
[2]: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=924573#15
[3]: https://fedora.pkgs.org/rawhide/fedora-x86_64/lcov-1.14-2.fc31.noarch.rpm.html

### CODE REVIEW

**Comment by AnAverageHuman** [comment]
The patch is too large to ship with the repo; consider packaging a [patch release](https://devmanual.gentoo.org/ebuild-writing/file-format/index.html) instead.

**Comment by juippis** [comment]
I'd prefer the proposed patch release as well, if at all possible. 
If not, I can upload the patch manually to mirrors.
(this should have a bug opened to block gcc-9 migration)

**Comment by gekola** [comment]
Updated to use Debian patches (this approach seems to be used for some other ebuilds).

**Comment by gekola** [comment]
Squashed and rebased the commits. Also fixed CI warnings (had to remove keywords not present in `dev-perl/PerlIO-gzip`).

File: dev-util/lcov/lcov-1.14_p2.ebuild [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+LCOV_V=${PV/_p*/}
+DB_V=${PV/*_p/}
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="https://github.com/linux-test-project/lcov.git"
+	inherit git-r3
+else
+	SRC_URI="
+		mirror://sourceforge/ltp/${PN}-${LCOV_V}.tar.gz
```
Comment (juippis) [inline dev-util/lcov/lcov-1.14_p2.ebuild]: Oh right, I dont remember whether this was killed or not. 
@mgorny any help here?

File: dev-util/lcov/lcov-1.14_p2.ebuild [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		mirror://debian/pool/main/l/${PN}/${PN}_${LCOV_V}-${DB_V}.debian.tar.xz
```
Comment (juippis) [inline dev-util/lcov/lcov-1.14_p2.ebuild]: Same here, let's wait for @mgorny to reply.

Looks good otherwise.

**Comment by zmedico** [comment]
I was looking into merging this, and then I encountered this unit test failure:
```
DATE .......: 2019-11-27 00:09:34 +0000
TESTNAME ...: lcov_diff_apply
COMMAND ....: "./diff_test"
OUTPUT .....: 
  Capturing coverage data from .
  Found gcov version: 9.2.0
  Using intermediate gcov format
  Scanning . for .gcda files ...
  Found 1 data files in .
  Processing prog.gcda
  Finished .info-file creation
  Capturing coverage data from .
  Found gcov version: 9.2.0
  Using intermediate gcov format
  Scanning . for .gcda files ...
  Found 1 data files in .
  Processing prog.gcda
  Finished .info-file creation
  Reading tracefile old/prog.info
  Reading diff diff
  Converting /var/tmp/portage/dev-util/lcov-1.14_p2/work/lcov-1.14/test/lcov_diff/old/prog.c
  1 entry converted, 0 entries left unchanged.
  Writing data to patched.info
  Summary coverage rate:
    lines......: 38.5% (5 of 13 lines)
    functions..: 66.7% (2 of 3 functions)
    branches...: 14.3% (1 of 7 branches)
  --- patched_normalized.info   2019-11-27 00:09:34.606941686 +0000
  +++ new_normalized.info       2019-11-27 00:09:34.600941686 +0000
  @@ -6,8 +6,6 @@
   FNDA:0,fn
   FNDA:1,fn2
   FNDA:1,main
  -FNF:3
  -FNH:2
   BRDA:6,0,0,-
   BRDA:6,0,1,-
   BRDA:6,0,2,-
  @@ -15,8 +13,6 @@
   BRDA:6,0,4,-
   BRDA:35,0,0,0
   BRDA:35,0,1,1
  -BRF:7
  -BRH:1
   DA:4,0
   DA:6,0
   DA:7,0
  @@ -30,6 +26,4 @@
   DA:35,1
   DA:36,0
   DA:38,1
  -LF:13
  -LH:5
   end_of_record
  Error: Mismatch in patched coverage data file
EXITCODE ...: 1
MEM ........: 87168kB
RESULT .....: fail
```

**Comment by zmedico** [comment]
Merged in 089acd4f622ce6a71cd4c8bcf438987a74d53ef4 with broken test disabled like this:
```
# Broken by https://github.com/linux-test-project/lcov/commit/75fbae1cfc5027f818a0bb865bf6f96fab3202da
rm -rf test/lcov_diff || die
```
Thanks!

**Pull request closed without merging**